js/src/logger.ts

```diff
@@ -4703,6 +4707,80 @@ export type PromptRowWithId<
     ? Pick<PromptRow, "_xact_id">
     : Partial<Pick<PromptRow, "_xact_id">>);
 
+export function deserializePlainStringAsJSON(s: string) {
+  if (s.trim() === "") {
+    return { value: null, error: undefined };
+  }
+
+  try {
+    return { value: JSON.parse(s), error: undefined };
+  } catch (e) {
+    return { value: s, error: e };
+  }
+}
+
+function renderTemplatedObject(obj: unknown, args: unknown): unknown {
```

Comment:
Maybe put these in a separate file for cleanliness?

Reply:
Is it worth it to have a separate file just for the render-related stuff?

Reply:
I figured if these functions are being used in other places then a separate file makes it somewhat clear that these are self-contained utils. Fine either way.